code/modules/mob/living/simple_animal/brainbot/brainbot.dm

```diff
@@ -0,0 +1,174 @@
+			qdel(src)
+		else
+			message_admins("Situation resolved! The brainbot is now under control of [src.key].")
+	src.mind.assigned_role = "mind control robot"
+	src.mind.special_role = "mind control robot"
+	serialNumber = rand(1,99999)
+	name = "MDR-[serialNumber]"
+	src << "<span class='boldannounce'>.://boot_seq-init</span>"
+	sleep(50)
+	src << "<span class='warning'>CYBERSUN INDUSTRIES MIND DOMINATION ROBOT S-[serialNumber] ONLINE.</span>"
+	sleep(30)
+	src << "<span class='warning'>SYSTEM CHECK REPORTS ALL SYSTEMS NOMINAL. CONNECTING TO COMMUNICATIONS MAINFRAME...</span>"
+	sleep(40)
+	src << "<span class='warning'>CONNECTION ESTABLISHED. ENABLING MOTOR FUNCTIONS...</span>"
+	sleep(40)
+	src << "<span class='warning'>MOTOR FUNCTIONS ESTABLISHED. RELINQUISHING REMOTE CONTROL TO MAIN UNIT. ESTABLISHING UNIQUE IDENTIFICATION...</span>"
+	src.notransform = 0
+	sleep(30)
+	var/list/possibleNames = list("CELL", "BADCELL", "JERK", "YOUNGLADY", "CREEP", "CHEERLEADER", "SNAPSHOT", "CLUCKER", "FETCH")
+	src.name = pick(possibleNames) + "-[serialNumber]"
+	src << "<span class='warning'>UNIQUE IDENTIFICATION CREATED. WELCOME, <b>[src.name]</b>. INITIATING GREETING SEQUENCE AND UNLOCKING FUNCTIONS...</span>"
+	sleep(30)
+	src << "<span class='info'>csOS v1.3: CYBERSUN INDUSTRIES MIND CONTROL UNIT S-[serialNumber]</span>"
+	src << "<span class='notice'>You are a Syndicate mind control robot manufactured by Cybersun Industries, a branch of the Syndicate.</span>"
+	src << "<span class='notice'>You have been deployed to [world.name] in order to harvest the memories of individuals.</span>"
+	src << "<span class='notice'>In order to do this, you must enter a host. This may be done by alt-clicking on a human.</span>"
+	src << "<span class='notice'>You have a variety of software you may use to accomplish this task. Good luck, S-[serialNumber]!</span>"
+	src << "<span class='warning'>...BOOT COMPLETE. Hello World!</span>"
+	src.say("Hello World!")
+	var/datum/objective/getMemories/O = new /datum/objective/getMemories()
+	O.owner = src
+	O.generateGoal()
+	src.mind.objectives += O
+	src.mind.current.verbs += /mob/living/simple_animal/brainbot/proc/ejectFromMob
+	src.mind.current.verbs += /mob/living/simple_animal/brainbot/proc/downloadMemory
+	src.mind.current.verbs += /mob/living/simple_animal/brainbot/proc/talkToHost
+	src.mind.current.verbs += /mob/living/simple_animal/brainbot/proc/helpHost
+	src.mind.current.verbs += /mob/living/simple_animal/brainbot/proc/harmHost
+	src.mind.current.verbs += /mob/living/simple_animal/brainbot/proc/silenceHost
+
+/mob/living/simple_animal/brainbot/Stat()
+	..()
+	if(statpanel("Status"))
+		stat(null, "MEMORIES DOWNLOADED: [storedMemories]TB")
+
+/mob/living/simple_animal/brainbot/Process_Spacemove(var/movement_dir = 0)
+	return 1 //Mainly to prevent the no-grav effect
+
+/mob/living/simple_animal/brainbot/suicide()
+	set hidden = 1
+	if(!canSuicide())
+		return
+	if(controlledMob)
```

Comment:
That makes sense.





On Fri, May 22, 2015 at 10:53 AM -0700, "Dorsisdwarf" <notifications@github.com> wrote:
> +     src.mind.current.verbs += /mob/living/simple_animal/brainbot/proc/harmHost
> +     src.mind.current.verbs += /mob/living/simple_animal/brainbot/proc/silenceHost
> +
> +/mob/living/simple_animal/brainbot/Stat()
> +     ..()
> +     if(statpanel("Status"))
> +             stat(null, "MEMORIES DOWNLOADED: [storedMemories]TB")
> +
> +/mob/living/simple_animal/brainbot/Process_Spacemove(var/movement_dir = 0)
> +     return 1 //Mainly to prevent the no-grav effect
> +
> +/mob/living/simple_animal/brainbot/suicide()
> +     set hidden = 1
> +     if(!canSuicide())
> +             return
> +     if(controlledMob)

That's because there's an "Are you sure". the user might become unable to suicide while the popup is active

---
Reply to this email directly or view it on GitHub:
https://github.com/tgstation/-tg-station/pull/9591/files#r30916530